src/core/gui/MainWindow.cpp

```diff
@@ -76,6 +77,14 @@ MainWindow::MainWindow(GladeSearchpath* gladeSearchPath, Control* control, GtkAp
     setSidebarVisible(control->getSettings()->isSidebarVisible());
 
     // Window handler
+    static constexpr auto interruptCallback = +[](gpointer data) -> gboolean {
+        if (interrupted()) {
+            auto control = static_cast<Control*>(data);
+            control->quit();
+        }
+        return true;
+    };
+    g_idle_add(static_cast<GSourceFunc>(interruptCallback), static_cast<gpointer>(this->control));
```

Comment:
This does not feel like the right way to catch the SIGINT signal: for instance, it fails if a dialog is open when the signal is received.

Reply:
What happens and what do you expect? There is also no other correct way I can think of. The problem with our dialogs is, that they block the event loop, which is a deprecated behavior from gtk2. And the close request must be made from the event loop.

Reply:
> What happens and what do you expect?

Nothing happens (besides the terminal outputing `Gracefully close requested with signal 2`).
I'd expect the app to close if the document has already been saved, or the "Save or Discard" dialog to pop up (on top or instead of the previously shown dialog). Or maybe to fallback on a disgraceful closure.

Reply:
Yes, that's kind of expected. When you press Ctrl+c a second time, xournalpp is closed disgracefully since the event loop could not handle the close request (in situations, where the event loop is blocked or the program hangs).

Note, that our possibilities are very limited in a signal handler. Every signal unsave function can introduce UB which is in any kind unwanted, when the user accidentally pressed ctrg+c.

Therefore I only set a flag which is then handled in the event loop.

Your wanted behavior should be introduced, when all dialogs get rid of the run function.